core-libs/setup/ssr/logger/loggers/express-server-logger.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Request } from 'express';
+import { LogContext, ServerLogger } from './server-logger';
+
+export class ExpressServerLogger extends ServerLogger {
+  log(message: string, context: LogContext): void {
+    /* eslint-disable-next-line no-console */
+    console.log(this.createLogMessage(message, context));
+  }
+  warn(message: string, context: LogContext): void {
+    /* eslint-disable-next-line no-console */
+    console.warn(this.createLogMessage(message, context));
+  }
+  error(message: string, context: LogContext): void {
+    /* eslint-disable-next-line no-console */
+    console.error(this.createLogMessage(message, context));
+  }
+
+  protected createLogMessage(message: string, context: LogContext): string {
+    const timestamp = new Date().toISOString();
+    const object = {
+      message,
+      context: {
+        timestamp,
+        ...context,
+      },
+    };
+    if (context.request) {
+      Object.assign(object.context, {
+        request: this.mapRequest(context.request),
+      });
+    }
+
+    return JSON.stringify(object);
+  }
+
+  protected mapRequest(request: Request): Record<string, any> {
+    return {
+      url: request.originalUrl,
```

Comment:
[MINOR, can be postponed]

Should we return only the URL path? Or also URL host (domain)? Or Both? IMHO both.

Why asking: because some customers might run a single NodeJS Spartacus instance to render multi-site (multiple basesites) hosted "virtually" in different domains. If both domains (e.g. `site-a.com` and `site-b.com`) have the same path URL (e.g. homepage '/' 😅), then those are 2 very different renderings, but they would appear with the same URL in kibana '/').

Note: we already have a function that we can use `getRequestOrigin(request)` - [source code](https://github.com/SAP/spartacus/blob/f3650ed2c6d05294f99012d8360ee7edbf4a0e8a/core-libs/setup/ssr/express-utils/express-request-origin.ts#L9-L26) - it returns the correct origin (domain), even if your NodeJS is behind a (chain of) reverse proxy.

```suggestion
      url: request.originalUrl,
      host: getRequestOrigin(request)
```

However I can see a downside of the above solution - it computes the origin from scratch for every logged message (instead of once per request).

Quick idea: pre-calculate the correct host once and then put it to `response.locals`. And then read it each time we log a message. 

We can postpone it to a different PR.

Reply:
sure, the link is here: https://jira.tools.sap/browse/CXSPA-3679